packages/core/src/external/carequality/pd/validating-pd.ts

```diff
@@ -0,0 +1,175 @@
+import {
+  Address as FHIRAddress,
+  ContactPoint,
+  Identifier,
+  Patient as FHIRPatient,
+} from "@medplum/fhirtypes";
+import { getStateEnum, USState } from "../../../domain/geographic-locations";
+import { Address } from "../../../domain/medical/address";
+import { Contact } from "../../../domain/medical/contact";
+import { PatientData, PersonalIdentifier } from "../../../domain/medical/patient";
+import { isContactType } from "../../fhir/patient";
+
+export class PatientAddressRequestedError extends Error {
+  constructor(message?: string) {
+    super(message);
+    this.name = "PatientAddressRequestedError";
+  }
+}
+
+export class LivingSubjectAdministrativeGenderRequestedError extends Error {
+  constructor(message?: string) {
+    super(message);
+    this.name = "LivingSubjectAdministrativeGenderRequestedError";
+  }
+}
+
+export class InternalError extends Error {
+  constructor(message?: string) {
+    super(message);
+    this.name = "InternalError";
+  }
+}
+
+export function validateFHIRAndExtractPatient(patient: FHIRPatient): PatientData {
```

Comment:
NABD now, but "validate" functions should validate as "far" as they can can, so whoever is calling/making use of the service doesn't have to send multiple requests/calls to learn about all the errors.

In this case, could have an array of issues and have each validation add to that array, error'ing at the end if the array is not empty.

But, unless we can report multiple errors to the IHE GW, I think it's fine to keep as is.

Reply:
I believe we can report multiple errors, by just adding multiple issues to the issues array. I am going to resolve this for now and add it to the todo list in the downstream PR to do it there, since will be easier to do there (can do this for pd, dq, dr all at once)